File: 181851.js
// 전국대회선발고사
function solution(rank, attendance) {
  for (let i = 0; i < attendance.length; i++) {
    if (attendance[i] == false) {
      rank.remove(i);
    }
  }
  rank.sort();
  return rank[0] * 10000 + rank[1] + 100 + rank[2];
}

console.log(
  solution([3, 7, 2, 5, 4, 6, 1], [false, true, true, true, true, false, false])
);

//다른사람풀이
// function solution(rank, attendance) {
//   const [a, b, c] = rank
//     .map((r, i) => [r, i])
//     .filter(([_, i]) => attendance[i])
//     .sort(([a], [b]) => a - b);
//   return 10000 * a[1] + 100 * b[1] + c[1];
// }

File: 181851.py
# 전국대회선발고사

def solution(rank,attendance):
    can_attendant = filter(lambda x: attendance[rank.index(x)] != False, rank)
    can_attendant =  sorted(list(can_attendant))
    return can_attendant[0] * 10000 + can_attendant[1] * 100 + can_attendant[2]
print(solution([3, 7, 2, 5, 4, 6, 1],[False, True, True, True, True, False, False]))